### PR TITLE
adding the option to setup the upstream HashiCorp repository

### PR DESCRIPTION
**Pull request by attachmentgenie**
HashiCorp now provides (some) upstream packages, this PR implements the managing of the repo

### CODE REVIEW

**Comment by solarkennedy** [comment]
I don't think we should default this to be true

**Comment by attachmentgenie** [comment]
@solarkennedy not a problem, i pushed the changes requested

File: manifests/install.pp [hunk 1]
```diff
@@ -65,6 +65,9 @@
       }
     }
     'package': {
+      if $consul::manage_repo{
+        include hashi_stack::repo
```
Comment (bastelfreak) [inline manifests/install.pp:69]: Should we also add this?
```puppet
Class['hashi_stack::repo'] -> Package[ $consul::package_name]
```

Comment (attachmentgenie) [reply manifests/install.pp:69]: makes sense, just pushed the update

**Review by bastelfreak — APPROVED** [review]

**Pull request merged**